common/utils.py

```diff
@@ -208,12 +208,22 @@ def inspect_directory(files: List[str], pattern: Optional[str] = None,
         else:
             continue
 
-def compile_code():
+def compile_code(makefile_target: str = ""):
     """Compiles the current directory (either with Make or manually)"""
-    ls_output = os.listdir()
-    if "Makefile" not in ls_output:
-        # Let's let the grader figure it out
-        os.system("bash")
+    if makefile_target == "":
```

Comment:
Sorry, maybe I could've added more detail to the issue. The default workflow here should be that if no makefile target is specified, we assume the default target of the Makefile (e.g. analogous to just running `make`).

Only drop into a shell if there is no Makefile in the current directory

Reply:
Got it - I was way over-complicating things.